packages/date/src/index.js

```diff
@@ -136,11 +136,34 @@ let settings = {
 export function setSettings( dateSettings ) {
 	settings = dateSettings;
 
-	// Backup and restore current locale.
+	setupWPTimezone();
+
+	// Does moment already have a locale with the right name?
+	if ( momentLib.locales().includes( dateSettings.l10n.locale ) ) {
+		// Is that locale misconfigured, e.g. because we are on a site running
+		// WordPress < 6.0?
```

Comment:
Noting to myself that I need to submit a Core patch that makes this same fix there. Should also add a comment here to delete this code when the minimum WP version is 6.0.